include/net/nrf_cloud_pgps.h

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef NRF_CLOUD_PGPS_H_
+#define NRF_CLOUD_PGPS_H_
+
+/** @file nrf_cloud_pgps.h
+ * @brief Module to provide nRF Cloud P-GPS support to nRF9160 SiP.
+ */
+
+#include <zephyr.h>
+#include <drivers/gps.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @defgroup nrf_cloud_pgps nRF Cloud P-GPS
+ * @{
+ */
+
+struct nrf_cloud_pgps_prediction;
+
+/** @brief Cell-based location request type */
+struct gps_pgps_request {
+	uint16_t prediction_count;
+	uint16_t prediction_period_min;
+	uint16_t gps_day;
+	uint32_t gps_time_of_day;
+} __packed;
+
+#define EAPPROXIMATE	8000 /* current time unknown; using first prediction */
+#define ELOADING	8001 /* not found but loading in progress */
+#define NRF_CLOUD_PGPS_EMPTY_EPHEM_HEALTH (0xff)
+
+enum nrf_cloud_pgps_event {
+	PGPS_EVT_INIT,
+	PGPS_EVT_UNAVAILABLE,
+	PGPS_EVT_LOADING,
+	PGPS_EVT_AVAILABLE,
+	PGPS_EVT_READY
+};
+
+typedef void (*pgps_event_handler_t)(enum nrf_cloud_pgps_event event,
+				     struct nrf_cloud_pgps_prediction *p);
+
+/**@brief Update storage of the most recent known location, in modem-specific
+ * normalized format (int32_t).
+ * Current time is also stored.
+ * Normalization:
+ *   (latitude / 90.0) * (1 << 23)
+ *   (longitude / 360.0) * (1 << 24)
+ *
+ * @param latitude Current latitude normalized.
+ * @param longitude Current longitude in normalized.
+ */
+void nrf_cloud_set_location_normalized(int32_t latitude, int32_t longitude);
+
+/**@brief Update the storage of the most recent known location in degrees.
+ * This will be injected along with the current time and relevant predicted
+ * ephemerii to the GPS unit in order to get the fastest possible fix.
+ * Current time is also stored.
+ *
+ * @param latitude Current latitude in degrees.
+ * @param longitude Current longitude in degrees.
+ */
+void nrf_cloud_set_location(double latitude, double longitude);
+
+/**@brief Update the storage of the leap second offset between GPS time
+ * and UTC.  Correct value for 2021, and default value, is 18.
```

Comment:
should the actual value be include here (in the function comments)?   it would need to be updated if `GPS_TO_UTC_LEAP_SECONDS` is changed.